src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -246,7 +246,7 @@ private function lexValue()
 
                     ++$this->cursor;
                 }
-                $value = rtrim($value);
+                $value = str_replace('\\\\', '\\', rtrim($value));
```

Comment:
Not sure that's enough: eg `FOO=A\\$B` should resolve to `A\Foo` when `B=Foo` is set previoulsy.

Reply:
you are right, and this also didn't work with double-quoted strings, both cases should be good now